File: SWIFT/src/Card.swift
/***************************************************************
* Name:      Console.swift
* Purpose:
*
* Contains information about a specific card.
* This class has memory optimization in mind as it
* refters to one common instance of artwork and name.
*
* Author:    Erik Welander (mail@erikwelander.se)
* Version:   2015-07-23
* Copyright: Erik Welander (https://www.facebook.com/erik.welander)
* License:   Creative Commons "by-nc-nd"
**************************************************************/
import Foundation

public class Card
{
    private var type: CardType = CardType.NOT_USED
    private var points: Int = 0
    init(type: CardType)
    {
        self.type = type
        self.points = 0
    }
    init(type: CardType, points: Int)
    {
        self.type = type
        self.points = points
    }
    public func getType() -> CardType
    {
        return self.type
    }
    public func getPoints() -> Int
    {
        return self.points
    }
    public func getArt() -> String
    {
        return CardArt.getCardArt(type)
    }
    public func getName() -> String
    {
        return CardManager.cardName[type.rawValue]
    }
}

File: JAVA/src/Card.java
/***************************************************************
 * Name:      Card.swift
 * Purpose:
 *
 * Contains information about a specific card.
 * This class has memory optimization in mind as it
 * refers to one common instance of artwork and name.
 *
 * Author:    Erik Welander (mail@erikwelander.se)
 * Version:   2015-07-30
 * Copyright: Erik Welander (https://www.facebook.com/erik.welander)
 * License:   Creative Commons "by-nc-nd"
 **************************************************************/
public class Card
{
    private CardType cType = CardType.NOT_USED;
    private int points = 0;

    Card(final CardType cType)
    {
        this.cType = cType;
        this.points = 0;
    }

    Card(final CardType cType, final int points)
    {
        this.cType = cType;
        this.points = points;
    }

    public final CardType getType()
    {
        return this.cType;
    }

    public final int getPoints()
    {
        return this.points;
    }

    public final String getArt()
    {
        return CardArt.getCardArt(this.cType);
    }

    public final String getName()
    {
        return this.cType.toString();
    }
}
